v2/controllers/marketplace/meterbase_controller.go

```diff
@@ -285,31 +379,81 @@ func (r *MeterBaseReconciler) Reconcile(request reconcile.Request) (reconcile.Re
 	// Install Objects
 	// ---
 
-	cfg := &corev1.Secret{}
-	prometheus := &monitoringv1.Prometheus{}
-	if result, _ := cc.Do(context.TODO(),
-		Do(r.reconcilePrometheusOperator(instance, factory)...),
-		Do(r.installMetricStateDeployment(instance, factory)...),
-		Do(r.reconcileAdditionalConfigSecret(cc, instance, prometheus, factory, cfg)...),
-		Do(r.reconcilePrometheus(instance, prometheus, factory, cfg)...),
-		Do(r.verifyPVCSize(reqLogger, instance, factory, prometheus)...),
-		Do(r.recyclePrometheusPods(reqLogger, instance, factory, prometheus)...),
-	); !result.Is(Continue) {
-		if result.Is(Error) {
-			reqLogger.Error(result, "error in reconcile")
-			return result.ReturnWithError(merrors.Wrap(result, "error creating prometheus"))
+	if userWorkloadMonitoringEnabled && transitionTimeExpired {
+		// Remove RHM Prom, transitionTime has expired
+		if result, _ := cc.Do(context.TODO(),
+			Do(r.uninstallPrometheusOperator(instance, factory)...),
+			Do(r.uninstallPrometheus(instance, factory)...),
+		); !result.Is(Continue) {
+			if result.Is(Error) {
+				reqLogger.Error(result, "error in reconcile")
+				return result.ReturnWithError(merrors.Wrap(result, "error uninstalling prometheus"))
+			}
+
+			reqLogger.Info("returing result", "result", *result)
+			return result.Return()
 		}
+	}
 
-		reqLogger.Info("returing result", "result", *result)
-		return result.Return()
+	promStsNamespacedName := types.NamespacedName{}
+	if userWorkloadMonitoringEnabled {
+		// Openshift provides Prometheus
+		if result, _ := cc.Do(context.TODO(),
+			Do(r.installPrometheusServingCertsCABundle(factory)...),
+			Do(r.installMetricStateDeployment(instance, factory)...),
+		); !result.Is(Continue) {
+			if result.Is(Error) {
+				reqLogger.Error(result, "error in reconcile")
+				return result.ReturnWithError(merrors.Wrap(result, "error creating metric-state"))
+			}
+
+			reqLogger.Info("returing result", "result", *result)
+			return result.Return()
+		}
+
+		promStsNamespacedName = types.NamespacedName{
+			Namespace: utils.OPENSHIFT_USER_WORKLOAD_MONITORING_NAMESPACE,
+			Name:      utils.OPENSHIFT_USER_WORKLOAD_MONITORING_STATEFULSET_NAME,
+		}
+	} else {
```

Comment:
```suggestion
	} else { // if userWorkload is not enabled
```

Reply:
Try to comment your else with the condition you are assuming in them. Helps us spot errors and remembering what code is doing.

Reply:
ok